diff --git a/lib/dvb/sec.cpp b/lib/dvb/sec.cpp
index d48d44e..e9a5d78 100644
--- a/lib/dvb/sec.cpp
+++ b/lib/dvb/sec.cpp
@@ -1005,6 +1005,9 @@ RESULT eDVBSatelliteEquipmentControl::clear()
 	for (eSmartPtrList<eDVBRegisteredFrontend>::iterator it(m_avail_frontends.begin()); it != m_avail_frontends.end(); ++it)
 	{
 		it->m_frontend->setData(eDVBFrontend::SATPOS_DEPENDS_PTR, -1);
+#ifdef BUILD_VUPLUS_DUO /* ikseong  */
+                it->m_frontend->setLoopthrough(0);
+#endif  
 		it->m_frontend->setData(eDVBFrontend::LINKED_PREV_PTR, -1);
 		it->m_frontend->setData(eDVBFrontend::LINKED_NEXT_PTR, -1);
 		it->m_frontend->setData(eDVBFrontend::ROTOR_POS, -1);
@@ -1015,6 +1018,9 @@ RESULT eDVBSatelliteEquipmentControl::clear()
 	for (eSmartPtrList<eDVBRegisteredFrontend>::iterator it(m_avail_simulate_frontends.begin()); it != m_avail_simulate_frontends.end(); ++it)
 	{
 		it->m_frontend->setData(eDVBFrontend::SATPOS_DEPENDS_PTR, -1);
+#ifdef BUILD_VUPLUS_DUO /* ikseong  */
+                it->m_frontend->setLoopthrough(0);
+#endif  
 		it->m_frontend->setData(eDVBFrontend::LINKED_PREV_PTR, -1);
 		it->m_frontend->setData(eDVBFrontend::LINKED_NEXT_PTR, -1);
 		it->m_frontend->setData(eDVBFrontend::ROTOR_POS, -1);
@@ -1435,6 +1441,9 @@ RESULT eDVBSatelliteEquipmentControl::setTunerLinked(int tu1, int tu2)
 			char c;
 			p1->m_frontend->setData(eDVBFrontend::LINKED_PREV_PTR, (long)p2);
 			p2->m_frontend->setData(eDVBFrontend::LINKED_NEXT_PTR, (long)p1);
+#ifdef BUILD_VUPLUS_DUO /* ikseong  */
+                        p1->m_frontend->setLoopthrough(1);
+#endif
 		}
 
 		p1=p2=NULL;
@@ -1450,6 +1459,9 @@ RESULT eDVBSatelliteEquipmentControl::setTunerLinked(int tu1, int tu2)
 		{
 			p1->m_frontend->setData(eDVBFrontend::LINKED_PREV_PTR, (long)p2);
 			p2->m_frontend->setData(eDVBFrontend::LINKED_NEXT_PTR, (long)p1);
+#ifdef BUILD_VUPLUS_DUO /* ikseong  */
+                        p1->m_frontend->setLoopthrough(1);
+#endif
 			return 0;
 		}
 	}
