Index: configure.ac
===================================================================
--- configure.ac	(revision 5)
+++ configure.ac	(revision 8)
@@ -221,11 +221,6 @@
 ac3lipsync/src/Makefile
 ac3lipsync/meta/Makefile
 
-dreamirc/Makefile
-dreamirc/src/Makefile
-dreamirc/src/protocols/Makefile
-dreamirc/meta/Makefile
-
 remotetimer/Makefile
 remotetimer/src/Makefile
 remotetimer/meta/Makefile
@@ -266,10 +261,6 @@
 podcast/src/Makefile
 podcast/meta/Makefile
 
-genuinedreambox/Makefile
-genuinedreambox/src/Makefile
-genuinedreambox/meta/Makefile
-
 kiddytimer/Makefile
 kiddytimer/img/Makefile
 kiddytimer/po/Makefile
@@ -281,16 +272,6 @@
 elektro/src/Makefile
 elektro/meta/Makefile
 
-ftpbrowser/Makefile
-ftpbrowser/po/Makefile
-ftpbrowser/src/Makefile
-ftpbrowser/src/images/Makefile
-ftpbrowser/meta/Makefile
-
-growlee/Makefile
-growlee/src/Makefile
-growlee/meta/Makefile
-
 meteoitalia/Makefile
 meteoitalia/src/Makefile
 
@@ -298,23 +279,8 @@
 weatherplugin/src/Makefile
 weatherplugin/meta/Makefile
 
-startupservice/Makefile
-startupservice/po/Makefile
-startupservice/src/Makefile
-startupservice/meta/Makefile
-
 movieselectionquickbutton/Makefile
 movieselectionquickbutton/po/Makefile
 movieselectionquickbutton/src/Makefile
 
-reconstructapsc/Makefile
-reconstructapsc/src_cc/Makefile
-reconstructapsc/src_py/Makefile
-reconstructapsc/meta/Makefile
-
-shoutcast/Makefile
-shoutcast/src/Makefile
-shoutcast/meta/Makefile
-shoutcast/po/Makefile
-
 ])
Index: Makefile.am
===================================================================
--- Makefile.am	(revision 5)
+++ Makefile.am	(revision 8)
@@ -5,17 +5,13 @@
 	autoresolution \
 	autotimer \
 	cdinfo \
-	dreamirc \
 	elektro \
 	emailclient \
 	epgrefresh \
 	epgsearch \
 	filebrowser \
 	fritzcall \
-	ftpbrowser \
-	genuinedreambox \
 	googlemaps \
-	growlee \
 	httpproxy \
 	imdb \
 	kiddytimer \
@@ -38,13 +34,10 @@
 	permanentclock \
 	podcast \
 	quickbutton \
-	reconstructapsc \
 	remotetimer \
 	rsdownloader \
 	simplerss \
 	setpasswd \
-	shoutcast \
-	startupservice \
 	startuptostandby \
 	tageditor \
 	trafficinfo \
