diff --git a/webinterface/src/plugin.py b/webinterface/src/plugin.py
index cbf42f8..caebbb0 100644
--- a/webinterface/src/plugin.py
+++ b/webinterface/src/plugin.py
@@ -228,7 +228,7 @@ def stopWebserver(session):
 # on given ipaddress, port, w/o auth, w/o ssl
 #===============================================================================
 def startServerInstance(session, ipaddress, port, useauth=False, l2k=None, usessl=False):
-	if hw.get_device_name().lower() != "dm7025":
+	if hw.get_device_name().lower() != "dm500hd":
 		l3k = None		
 		l3c = tpm.getCert(eTPM.TPMD_DT_LEVEL3_CERT)
 		
@@ -550,7 +550,7 @@ def checkBonjour():
 def networkstart(reason, **kwargs):
 	l2r = False
 	l2k = None
-	if hw.get_device_name().lower() != "dm7025":		
+	if hw.get_device_name().lower() != "dm500hd":		
 		l2c = tpm.getCert(eTPM.TPMD_DT_LEVEL2_CERT)
 		
 		if l2c is None:
@@ -579,7 +579,7 @@ def openconfig(session, **kwargs):
 def configCB(result, session):
 	l2r = False
 	l2k = None
-	if hw.get_device_name().lower() != "dm7025":		
+	if hw.get_device_name().lower() != "dm500hd":		
 		l2c = tpm.getCert(eTPM.TPMD_DT_LEVEL2_CERT)
 		
 		if l2c is None:
